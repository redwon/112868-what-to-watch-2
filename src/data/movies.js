const movies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    name: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`
  },
  {
    name: `Macbeth`,
    img: `img/macbeth.jpg`
  },
  {
    name: `Mindhunter`,
    img: `img/mindhunter.jpg`
  }
];

export default movies;
