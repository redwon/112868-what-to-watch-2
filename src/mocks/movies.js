const movies = [
  {
    id: 1,
    name: `War of the Worlds`,
    posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg`,
    previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg`,
    backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg`,
    backgroundColor: `#9B7E61`,
    description: `Lorem ipsum`,
    rating: 9.3,
    scoresCount: 386834,
    director: `Steven Spielberg`,
    starring: [`Tom Cruise`, `Dakota Fanning`, `Tim Robbins`],
    runTime: 116,
    genre: `Adventure`,
    released: 2005,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2,
    name: `Bronson`,
    posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/bronson.jpg`,
    previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/bronson.jpg`,
    backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/bronson.jpg`,
    backgroundColor: `#73B39A`,
    description: `Lorem ipsum`,
    rating: 3.6,
    scoresCount: 109661,
    director: `Nicolas Winding Refn`,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    runTime: 92,
    genre: `Action`,
    released: 2008,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

export default movies;
