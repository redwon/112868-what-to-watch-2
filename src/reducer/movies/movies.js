export const initialState = [];

export const ActionType = {
  LOAD_MOVIES: `LOAD_MOVIES`,
};

export const ActionCreator = {
  loadMovies: (movies) => ({
    type: ActionType.LOAD_MOVIES,
    payload: movies
  }),
};

export const Operations = {
  loadMovies: () => (dispatch, _, api) => {
    return api.get(`/films`)
      .then((response) => {
        dispatch(ActionCreator.loadMovies(response.data));
      });
  }
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_MOVIES:
      return action.payload;
  }

  return state;
};
