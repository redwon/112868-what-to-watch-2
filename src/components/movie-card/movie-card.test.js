import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/movies';

import MovieCard from './movie-card';

const createNodeMock = (element) => {
  if (element.type === `video`) {
    return {};
  }
  return null;
};

it(`renders correctly`, () => {
  const tree = renderer
    .create(<MovieCard
      movie={movies[0]}
      onClick={() => {}}
      onHover={() => {}}
    />, {createNodeMock})
    .toJSON();

  expect(tree).toMatchSnapshot();
});
