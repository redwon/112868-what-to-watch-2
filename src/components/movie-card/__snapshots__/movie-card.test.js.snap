// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      className="player__video"
      muted={true}
      poster="https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
      onClick={[Function]}
    >
      War of the Worlds
    </a>
  </h3>
</article>
`;
